### ISSUE TITLE
dense net training from scratch accuracy keeps dropping

### ISSUE THREAD

**Issue by cxy7452** [MaybeShewill-CV/lanenet-lane-detection#10]
Hi,

This is great, thanks for sharing the code!

I was using your code with processed TuSimple data trying to train dense net model from scratch according to your training procedure and config. While the training loss keeps decreasing, the accuracy also keeps decreasing (from around 0.5 at the beginning down to now close to 0.03 at the 500th epoch). I am using the same learning rate setup as your suggested setup. Any thoughts on why this is happening?

**Comment by MaybeShewill-CV** [comment]
@cxy7452 Since I have not find a suitable pre-trained weights on dense net yet all my experiments was done using the vgg structure as the base cnn net

**Comment by yiyichun** [comment]
Hello, @cxy7452 :
How do you train? I have a problem with nan .(https://github.com/MaybeShewill-CV/lanenet-lane-detection/issues/4)
Do you need to read the json file during training? Or just need train.txt 、 val.txt、binary segmentation label file 、 instance segmentation label file and the original image ?
thank you :)

**Comment by yiyichun** [comment]
Hi , @cxy7452:
Thank you for your reply.
My training data is as follows:
original image:
![1492626270684175793](https://user-images.githubusercontent.com/34088341/42306308-1bba4abe-8061-11e8-8d45-1ec8bf9295fc.jpg)

gt_image_binary:
![1492626270684175793](https://user-images.githubusercontent.com/34088341/42306335-28955b16-8061-11e8-9f89-ad843bd9872c.jpg)

gt_image_instance:
![1492626270684175793](https://user-images.githubusercontent.com/34088341/42306343-3359d1f8-8061-11e8-8f7a-538878d98db0.jpg)

The path of the above three is recorded in train.txt
Is this correct? thank you :)

**Comment by MaybeShewill-CV** [comment]
@yiyichun The label image looks correct for now. You'd better check the label follow the readme.md file

**Comment by yiyichun** [comment]
Hi, @MaybeShewill-CV ,
Does the image file need png file? Is the jpg image file available?

**Comment by MaybeShewill-CV** [comment]
@yiyichun jpg file is also accepted. But you have to make sure the pixel value of the label image is right.

**Comment by ding-hai-tao** [comment]
Hi,@yiyichun ,
How is your gt_image_binary and gt_iamge_instance generated?

**Comment by yiyichun** [comment]
Hi, @ding-hai-tao ,
I am modifying data_provider\lanenet_hnet_data_processor.py to get these 2 files.
However, I am not sure if this is correct, maybe @MaybeShewill-CV  can have correct instructions.

**Comment by south-ocean** [comment]
Hello, I also meet the problem that dense net training from scratch accuracy keeps dropping , can you share some experience? @cxy7452 @MaybeShewill-CV @yiyichun @ding-hai-tao

**Comment by MaybeShewill-CV** [comment]
@south-ocean You'd better use pretrained dense net weights params as initialized weights.

**Comment by yiyichun** [comment]
@south-ocean ,
I have also encountered the same problem. Try to reduce the learning rate.

**Comment by south-ocean** [comment]
Your solution didn't resolve my problem, can you leave me a Contact？ Thank  you! @yiyichun @MaybeShewill-CV @cxy7452

**Comment by MaybeShewill-CV** [comment]
@south-ocean You can try this if the solution can resolve your problem. #33  @mukaman84

**Comment by south-ocean** [comment]
@MaybeShewill-CV  @yiyichun  @cxy7452 I have decreased my initial learning rate to 0.00005 and 0.0002, but it also did't work. I train the network on culane dataset and base on the vgg16.py.  My  result  is up and down just as the picture show. Can you have some suggestion. I think this is a stranger phenomenon.
I0730 20:56:55.060886 32051 train_lanenet.py:161] Global configuration is as follows:

I0730 20:56:55.061288 32051 train_lanenet.py:162] {'TEST': {'BATCH_SIZE': 32, 'GPU_MEMORY_FRACTION': 0.8, 'TF_ALLOW_GROWTH': True}, 'TRAIN': {'IMG_WIDTH': 512, 'DISPLAY_STEP': 5, 'TF_ALLOW_GROWTH': True, 'LEARNING_RATE': 2e-05, 'IMG_HEIGHT': 256, 'MOMENTUM': 0.9, 'LR_DECAY_RATE': 0.1, 'BATCH_SIZE': 4, 'LR_DECAY_STEPS': 210000, 'TEST_DISPLAY_STEP': 1000, 'EPOCHS': 200010, 'CLASSES_NUMS': 2, 'GPU_MEMORY_FRACTION': 0.85, 'VAL_BATCH_SIZE': 4}}
I0730 20:56:56.034889 32051 train_lanenet.py:170] Training from scratch
I0730 20:57:03.815589 32051 train_lanenet.py:292] Epoch: 1 total_loss= 3.182313 binary_seg_loss= 0.772669 instance_seg_loss= 8.804817 accuracy= 0.558625 mean_cost_time= 4.906305s 
I0730 20:57:03.816108 32051 train_lanenet.py:300] Epoch_Val: 1 total_loss= 3140.431641 binary_seg_loss= 49.420410 instance_seg_loss= 10352.791016 accuracy= 0.518704 mean_cost_time= 0.337986s 
I0730 20:57:13.789867 32051 train_lanenet.py:292] Epoch: 6 total_loss= 2.984972 binary_seg_loss= 0.665445 instance_seg_loss= 8.397202 accuracy= 0.411133 mean_cost_time= 1.221524s 
I0730 20:57:22.539421 32051 train_lanenet.py:292] Epoch: 11 total_loss= 2.722105 binary_seg_loss= 0.569822 instance_seg_loss= 7.744099 accuracy= 0.291561 mean_cost_time= 1.200509s 
I0730 20:57:31.513262 32051 train_lanenet.py:292] Epoch: 16 total_loss= 2.631841 binary_seg_loss= 0.499258 instance_seg_loss= 7.607867 accuracy= 0.190310 mean_cost_time= 1.221127s 
I0730 20:57:40.639294 32051 train_lanenet.py:292] Epoch: 21 total_loss= 2.037402 binary_seg_loss= 0.457812 instance_seg_loss= 5.723112 accuracy= 0.168230 mean_cost_time= 1.236754s 
I0730 20:57:49.710389 32051 train_lanenet.py:292] Epoch: 26 total_loss= 1.894807 binary_seg_loss= 0.407035 instance_seg_loss= 5.366275 accuracy= 0.137988 mean_cost_time= 1.227352s 
I0730 20:57:58.873020 32051 train_lanenet.py:292] Epoch: 31 total_loss= 1.433069 binary_seg_loss= 0.401858 instance_seg_loss= 3.839229 accuracy= 0.177701 mean_cost_time= 1.239474s 
I0730 20:58:08.021443 32051 train_lanenet.py:292] Epoch: 36 total_loss= 1.342402 binary_seg_loss= 0.352276 instance_seg_loss= 3.652697 accuracy= 0.101736 mean_cost_time= 1.237862s 
I0730 20:58:17.079957 32051 train_lanenet.py:292] Epoch: 41 total_loss= 1.097423 binary_seg_loss= 0.319540 instance_seg_loss= 2.912483 accuracy= 0.188466 mean_cost_time= 1.235295s 
I0730 20:58:26.174401 32051 train_lanenet.py:292] Epoch: 46 total_loss= 1.052523 binary_seg_loss= 0.302006 instance_seg_loss= 2.803729 accuracy= 0.117257 mean_cost_time= 1.243806s 
I0730 20:58:35.147967 32051 train_lanenet.py:292] Epoch: 51 total_loss= 0.917678 binary_seg_loss= 0.261368 instance_seg_loss= 2.449069 accuracy= 0.217791 mean_cost_time= 1.217244s 
I0730 20:58:44.348255 32051 train_lanenet.py:292] Epoch: 56 total_loss= 0.787788 binary_seg_loss= 0.275637 instance_seg_loss= 1.982805 accuracy= 0.090680 mean_cost_time= 1.248770s 
I0730 20:58:53.217406 32051 train_lanenet.py:292] Epoch: 61 total_loss= 0.818690 binary_seg_loss= 0.226270 instance_seg_loss= 2.201004 accuracy= 0.170645 mean_cost_time= 1.228045s 
I0730 20:59:02.397592 32051 train_lanenet.py:292] Epoch: 66 total_loss= 0.818188 binary_seg_loss= 0.202553 instance_seg_loss= 2.254669 accuracy= 0.098575 mean_cost_time= 1.242261s 
I0730 20:59:11.563933 32051 train_lanenet.py:292] Epoch: 71 total_loss= 0.751750 binary_seg_loss= 0.187893 instance_seg_loss= 2.067414 accuracy= 0.082230 mean_cost_time= 1.245210s 
I0730 20:59:20.727879 32051 train_lanenet.py:292] Epoch: 76 total_loss= 0.691930 binary_seg_loss= 0.172193 instance_seg_loss= 1.904650 accuracy= 0.033963 mean_cost_time= 1.242907s 
I0730 20:59:29.823996 32051 train_lanenet.py:292] Epoch: 81 total_loss= 0.938171 binary_seg_loss= 0.201278 instance_seg_loss= 2.657588 accuracy= 0.021553 mean_cost_time= 1.229116s 
I0730 20:59:39.008086 32051 train_lanenet.py:292] Epoch: 86 total_loss= 0.733594 binary_seg_loss= 0.170314 instance_seg_loss= 2.047914 accuracy= 0.011249 mean_cost_time= 1.247131s 
I0730 20:59:48.166474 32051 train_lanenet.py:292] Epoch: 91 total_loss= 0.856981 binary_seg_loss= 0.169592 instance_seg_loss= 2.460891 accuracy= 0.002330 mean_cost_time= 1.241629s 
I0730 20:59:57.302298 32051 train_lanenet.py:292] Epoch: 96 total_loss= 0.726884 binary_seg_loss= 0.203124 instance_seg_loss= 1.948992 accuracy= 0.003665 mean_cost_time= 1.244528s 
I0730 21:00:06.454951 32051 train_lanenet.py:292] Epoch: 101 total_loss= 0.733755 binary_seg_loss= 0.188203 instance_seg_loss= 2.006709 accuracy= 0.005941 mean_cost_time= 1.239348s 
I0730 21:00:15.563410 32051 train_lanenet.py:292] Epoch: 106 total_loss= 0.663975 binary_seg_loss= 0.173088 instance_seg_loss= 1.809376 accuracy= 0.003644 mean_cost_time= 1.242064s 
I0730 21:00:24.603786 32051 train_lanenet.py:292] Epoch: 111 total_loss= 0.679454 binary_seg_loss= 0.177563 instance_seg_loss= 1.850531 accuracy= 0.004317 mean_cost_time= 1.223098s 
I0730 21:00:33.728395 32051 train_lanenet.py:292] Epoch: 116 total_loss= 0.552807 binary_seg_loss= 0.161086 instance_seg_loss= 1.466822 accuracy= 0.004220 mean_cost_time= 1.230377s 
I0730 21:00:42.719569 32051 train_lanenet.py:292] Epoch: 121 total_loss= 0.608239 binary_seg_loss= 0.153512 instance_seg_loss= 1.669267 accuracy= 0.002042 mean_cost_time= 1.219675s 
I0730 21:00:51.751444 32051 train_lanenet.py:292] Epoch: 126 total_loss= 0.562054 binary_seg_loss= 0.179836 instance_seg_loss= 1.453897 accuracy= 0.008693 mean_cost_time= 1.220884s 
I0730 21:01:00.919986 32051 train_lanenet.py:292] Epoch: 131 total_loss= 0.621467 binary_seg_loss= 0.178786 instance_seg_loss= 1.654391 accuracy= 0.017167 mean_cost_time= 1.242262s 
I0730 21:01:10.064341 32051 train_lanenet.py:292] Epoch: 136 total_loss= 0.678088 binary_seg_loss= 0.172225 instance_seg_loss= 1.858435 accuracy= 0.003781 mean_cost_time= 1.240239s 
I0730 21:01:19.229028 32051 train_lanenet.py:292] Epoch: 141 total_loss= 0.444269 binary_seg_loss= 0.144868 instance_seg_loss= 1.142870 accuracy= 0.001411 mean_cost_time= 1.239088s 
I0730 21:01:28.381182 32051 train_lanenet.py:292] Epoch: 146 total_loss= 0.447428 binary_seg_loss= 0.172033 instance_seg_loss= 1.090017 accuracy= 0.013723 mean_cost_time= 1.236260s
